lib/Support/CompilationLog.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Support/CompilationLog.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/FileSystem.h"
+
+#include <fstream>
+
+namespace glow {
+llvm::cl::opt<bool>
+    dumpCompilationLogOpt("dump-compilation-log",
+                     llvm::cl::desc("Dump compilation log"));
+
+std::vector<std::string> CompilationScope::scopeStack = {};
+
+CompilationScope::CompilationScope(std::string scopeName) {
+    this->scopeName = scopeName;
+    scopeStack.push_back(scopeName);
+}
+
+CompilationScope::~CompilationScope() {
+    scopeStack.pop_back();
+}
+
+llvm::StringRef CompilationScope::getCurrentCompilationScopeStack() {
+    std::string scopes = "";
+    for (auto scopeStr : scopeStack) {
```

Comment:
You can optimize this a bit if you create a member variable for the full scope name and update it when a `CompilationScope` is created or  destroyed. If this is called multiple times within the same scope it will redo the same work each time to get the full scope name.

Reply:
added!